allennlp/training/trainer.py

```diff
@@ -258,6 +259,8 @@ def _train_epoch(self, epoch: int) -> dict:
                 logger.info("Batch %d/%d: %s", batch_num, num_training_batches, description)
                 self._last_log = time.time()
 
+        time.strftime("Duration: %H:%M:%S", time.gmtime(elapsed_time))
```

Comment:
`tqdm` gives you durations already, so this should only be printed in the `no_tqdm` case.